pkgs/tools/security/pass/default.nix

```diff
@@ -13,12 +14,14 @@
 with lib;
 
 assert x11Support -> xclip != null;
+assert waylandSupport -> wl-clipboard != null;
 
-assert dmenuSupport -> dmenu != null
-                       && xdotool != null
-                       && x11Support;
+assert dmenuSupport -> x11Support || waylandSupport;
+assert dmenuSupport && x11Support
+  -> dmenu != null && xdotool != null;
+assert dmenuSupport && waylandSupport
+  -> dmenu-wayland != null && ydotool != null;
```

Comment:
Can we stop adding new asserts and ``? null`` like this? If you remove the ``? null from inputs we don't need the assert because nixpkgs actually checks if a package is null. In this usecase it is just extra syntactic sugar which breaks some overwrites.

Reply:
Some of the asserts are strictly necessary, but we can probably get rid of the `? null`s indeed.